src/Symfony/Component/Amqp/Queue.php

```diff
@@ -0,0 +1,142 @@
+     * * retry_strategy: A retry strategy instance to use (see RetryStrategyInterface)
+     * * retry_strategy_queue_pattern:
+     *   * The queue pattern to use for messages that needs to wait (default to %exchange%.%time%.wait)
+     *   * The pattern is expanded according to the %exchange% and %time% values.
+     * * bind_arguments: An array of bind arguments
+     */
+    public function __construct(\AMQPChannel $channel, $name, array $arguments = array(), $declare = true)
+    {
+        $this->bindings = array();
+
+        parent::__construct($channel);
+
+        $this->setName($name);
+
+        if (array_key_exists('routing_keys', $arguments)) {
+            $routingKeys = $arguments['routing_keys'];
+            if (is_string($routingKeys)) {
+                $routingKeys = array($routingKeys);
+            }
+            if (!is_array($routingKeys) && null !== $routingKeys && false !== $routingKeys) {
+                throw new \InvalidArgumentException(sprintf('"routing_keys" option should be a string, false, null or an array of string, "%s" given.', gettype($routingKeys)));
+            }
+
+            unset($arguments['routing_keys']);
+        } else {
+            $routingKeys = array($name);
+        }
+
+        if (isset($arguments['flags'])) {
+            $this->setFlags($arguments['flags']);
+            unset($arguments['flags']);
+        } else {
+            $this->setFlags(\AMQP_DURABLE);
+        }
+
+        if (isset($arguments['exchange'])) {
+            $exchange = $arguments['exchange'];
+            unset($arguments['exchange']);
+        } else {
+            $exchange = Broker::DEFAULT_EXCHANGE;
+        }
+
+        if (array_key_exists('retry_strategy', $arguments)) {
+            $this->retryStrategy = $arguments['retry_strategy'];
+            if (!$this->retryStrategy instanceof RetryStrategyInterface) {
+                throw new \InvalidArgumentException('The retry_strategy should be an instance of RetryStrategyInterface.');
+            }
+            unset($arguments['retry_strategy']);
+        }
+
+        if (array_key_exists('retry_strategy_queue_pattern', $arguments)) {
+            $this->retryStrategyQueuePattern = $arguments['retry_strategy_queue_pattern'];
+            unset($arguments['retry_strategy_queue_pattern']);
+        } else {
+            $this->retryStrategyQueuePattern = '%exchange%.%time%.wait';
+        }
+
+        if (isset($arguments['bind_arguments'])) {
+            $bindArguments = $arguments['bind_arguments'];
+            unset($arguments['bind_arguments']);
+        } else {
+            $bindArguments = array();
+        }
+
+        $this->setArguments($arguments);
+
+        if (null === $routingKeys) {
+            $this->bindings[$exchange][] = array(
+                'routing_key' => $routingKeys,
+                'bind_arguments' => $bindArguments,
+            );
+        } elseif (is_array($routingKeys)) {
```

Comment:
You mean It could be just a `if` instead of an `elseif` ? 
If yes, what's the benefit?